### PR TITLE
feat(cli): add initial CLI support for object manifests

### PR DESCRIPTION
**Pull request by Avantol13**
### New Features
- objects CLI with support for calling functions to manage file object manifests (functions already available in SDK)

### CODE REVIEW

File: gen3/cli/objects.py [hunk 1]
```diff
@@ -239,7 +241,59 @@ def objects_manifest_publish(ctx, file, thread_num, append_urls):
     )
 
 
+@click.command(help="Publishes specified object manifest to Gen3 instance.")
```
Comment (paulineribeyre) [inline gen3/cli/objects.py:244]: the description should be updated

File: gen3/cli/objects.py [hunk 2]
```diff
@@ -239,7 +241,59 @@ def objects_manifest_publish(ctx, file, thread_num, append_urls):
...
+@click.argument("file", required=True)
+@click.pass_context
+def objects_manifest_delete_all_guids(ctx, file):
+    auth = ctx.obj["auth_factory"].get()
+    loop = asyncio.get_event_loop()
+
+    if not file:
+        file = click.prompt("Enter Discovery metadata file path to publish")
```
Comment (paulineribeyre) [inline gen3/cli/objects.py:252]: here too ("publish")

File: gen3/cli/objects.py [hunk 3]
```diff
@@ -239,7 +241,59 @@ def objects_manifest_publish(ctx, file, thread_num, append_urls):
...
+    )
+
+    index = Gen3Index(auth.endpoint, auth_provider=auth)
+    if not index.is_healthy():
+        print(
+            f"uh oh! The indexing service is not healthy in the commons {auth.endpoint}"
+        )
+        exit()
+
+    # if delimter not specified, try to get based on file ext
```
Comment (paulineribeyre) [inline gen3/cli/objects.py:271]:
```suggestion
    # if delimiter not specified, try to get based on file ext
```

File: gen3/cli/objects.py [hunk 4]
```diff
@@ -239,7 +241,59 @@ def objects_manifest_publish(ctx, file, thread_num, append_urls):
...
+    file_ext = os.path.splitext(file)
```
Comment (paulineribeyre) [inline gen3/cli/objects.py:272]: "if delimiter not specified": maybe you meant to accept the delimiter as an argument to the command?

File: gen3/cli/objects.py [hunk 5]
```diff
@@ -239,7 +241,59 @@ def objects_manifest_publish(ctx, file, thread_num, append_urls):
...
+        csvReader = csv.DictReader(input_file, delimiter=manifest_file_delimiter)
+        fieldnames = csvReader.fieldnames
+
+        logging.debug(f"got fieldnames from {file}: {fieldnames}")
+        logging.debug(f"using guid/GUID/did/DID to retrieve GUID to delete...")
+
+        for row in csvReader:
+            guid = (
+                row.get("guid") or row.get("GUID") or row.get("did") or row.get("DID")
+            )
```
Comment (paulineribeyre) [inline gen3/cli/objects.py:289]: would it be more performant to only check this once? we could check which name is used in `fieldnames`

File: gen3/cli/objects.py [hunk 6]
```diff
@@ -239,7 +241,59 @@ def objects_manifest_publish(ctx, file, thread_num, append_urls):
...
+            if guid:
+                print(f"deleting GUID record:{guid}")
+                print(index.delete_record(guid=guid))
```
Comment (paulineribeyre) [inline gen3/cli/objects.py:292]:
```suggestion
                logging.debug(f"deleting GUID record:{guid}")
                logging.debug(index.delete_record(guid=guid))
```

File: gen3/cli/objects.py [hunk 7]
```diff
@@ -239,7 +241,59 @@ def objects_manifest_publish(ctx, file, thread_num, append_urls):
...
+    )
+    click.confirm(
+        f"Please confirm again, this is irreversible. All GUIDs specified specified in {file} will be deleted from {auth.endpoint}. You are sure?",
+        abort=True,
+    )
+
+    index = Gen3Index(auth.endpoint, auth_provider=auth)
+    if not index.is_healthy():
+        print(
+            f"uh oh! The indexing service is not healthy in the commons {auth.endpoint}"
```
Comment (paulineribeyre) [inline gen3/cli/objects.py:267]:
```suggestion
        logging.error(
            f"uh oh! The indexing service is not healthy in the commons {auth.endpoint}"
```

File: gen3/cli/objects.py [hunk 8]
```diff
@@ -239,7 +241,59 @@ def objects_manifest_publish(ctx, file, thread_num, append_urls):
...
+
+    # if delimter not specified, try to get based on file ext
+    file_ext = os.path.splitext(file)
+    if file_ext[-1].lower() == ".tsv":
+        manifest_file_delimiter = "\t"
+    else:
+        # default, assume CSV
+        manifest_file_delimiter = ","
+
+    with open(file, "r", encoding="utf-8-sig") as input_file:
```
Comment (paulineribeyre) [inline gen3/cli/objects.py:279]: i think we should move this functionality to an SDK function, and have the CLI call it. right now it's available in the CLI and not in the SDK

File: gen3/cli/objects.py [hunk 9]
```diff
@@ -25,50 +25,60 @@ def manifest():
...
 )
 @click.pass_context
-def objects_manifest_read(ctx, num_processes, max_concurrent_requests):
+def objects_manifest_read(ctx, output_file, num_processes, max_concurrent_requests):
     auth = ctx.obj["auth_factory"].get()
     loop = asyncio.get_event_loop()
     click.echo(f"Getting minimal object metadata from {auth.endpoint}")
-    output_file = loop.run_until_complete(
+    output_filename = loop.run_until_complete(
         indexing.async_download_object_manifest(
```
[the remark this answers is not in the document]
Comment (paulineribeyre) [reply gen3/cli/objects.py]: still wondering about this - does `click.echo(output_filename)` on line 64 just output "None"?

File: gen3/cli/objects.py (hunk 8 above)
Comment (Avantol13) [reply gen3/cli/objects.py:279]: good point, will do

File: gen3/cli/objects.py (hunk 9 above)
[the remark this answers is not in the document]
Comment (Avantol13) [reply gen3/cli/objects.py]: oh whoops, missed this. will adjust

**Review by paulineribeyre — APPROVED** [review]

**Pull request merged**